modules/fedimint-ln/src/lib.rs

```diff
@@ -796,16 +791,15 @@ impl ServerModulePlugin for LightningModule {
             },
             api_endpoint! {
                 "/list_gateways",
-                async |module: &LightningModule, _v: ()| -> Vec<LightningGateway> {
-                    Ok(module.list_gateways(&module.non_consensus_db.begin_transaction(module.decoders.clone())))
+                async |module: &LightningModule, dbtx, _v: ()| -> Vec<LightningGateway> {
+                    Ok(module.list_gateways(&dbtx))
                 }
             },
             api_endpoint! {
                 "/register_gateway",
-                async |module: &LightningModule, gateway: LightningGateway| -> () {
-                    futures::executor::block_on( async {
-                       module.register_gateway(gateway).await;
-                    });
+                async |module: &LightningModule, dbtx, gateway: LightningGateway| -> () {
+                    module.register_gateway(&mut dbtx, gateway).await;
+                    dbtx.commit_tx().await.expect("DB Error");
```

Comment:
Perhaps not in this PR, but now that the commit_tx has bubbled up to the api endpoint, should we return error to the user if this fails? (it could potentially happen if the disk is full or something)